lib/rubocop/cop/lint/boolean_symbol.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for boolean symbols.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   :true
+      #
+      # @example
+      #
+      #   # good
+      #
+      #   true
+      #
+      class BooleanSymbol < Cop
+        MSG = 'Boolean symbol - you probably meant to use `%s`.'.freeze
+
+        def on_sym(node)
+          match = node.source.match(/\A:(true|false)\z/)
```

Comment:
That seems like an overkill - you can just check if the symbol is `:true` or `:false`.